src/Schema.ts

```diff
@@ -277,6 +277,61 @@ class Schema<
 		return [...positions];
 	}
 
+	/** Transform the paths to ordinal positions. Returning a '.' delimited string. Ex. "1.2.3"  */
+	public transformPathsToOrdinalPositions(paths: string[]): string[] {
+		if (paths.length === 0) {
+			return [];
+		}
+
+		const positionPaths = this.getPositionPaths();
+		const positionKeys = Array.from(positionPaths.keys());
+
+		const positions = paths.reduce((acc, positionPath) => {
+			if (positionPaths.has(positionPath)) {
+				// find the key in position paths
+				// add position
+				const ordinalPath =
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					positionPaths.get(positionPath)!;
+				acc.add(this.buildOrdinalPathString(ordinalPath));
+			} else if (!positionPath.includes('.')) {
+				// if the property is a parent key, we will add positions for all children
+				// e.g we only pass property "name" to return all data for name.first, name.last, etc.
+				const matchedPositionPaths = positionKeys.filter(
+					(key) => key.split('.')[0] === positionPath,
+				);
+				matchedPositionPaths.forEach((key) => {
+					// add child position
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					const ordinalPath = positionPaths.get(key)!;
+					acc.add(this.buildOrdinalPathString(ordinalPath));
+				});
+			}
+			return acc;
+		}, new Set<string>());
+
+		return [...positions];
+	}
+
+	private buildOrdinalPathString(ordinalPath: number[]): string {
```

Comment:
This feels like we're relying on a quirk of UniBasic where it will let you have empty strings for values and subvalues. It also means this code is more complex than it needs to be. It could simply be:

```
		const [attributePosition, valuePosition = 0, subvaluePosition = 0] = ordinalPath;
		return `${attributePosition + 1}.${valuePosition + 1}.${subvaluePosition + 1}`;
```

That code is then simple enough that it can probably just be consolidated into the `transformPathsToOrdinalPositions` method (especially since I think that can be slimmed down) and it won't need additional checks for length and such.

Reply:
Consolidated into `transformPathsToOrdinalPositions`